examples/app-api/app/request-handler.ts

```diff
@@ -0,0 +1,69 @@
+import { NextRequest, NextResponse } from "next/server";
+
+type RouteHandler<T = Record<string, string | string[]>> = (
+  req: NextRequest,
+  ctx: { params: T },
+) => Promise<Response> | Response;
+
+function isAllowedOrigin(req: NextRequest): boolean {
+  const referer = req.headers.get("referer");
+  const host = req.headers.get("host");
+
+  // Allow requests from the same origin
+  if (referer && host) {
+    const refererUrl = new URL(referer);
+    return refererUrl.host === host;
+  }
+
+  if (host?.includes("localhost")) {
+    return true;
+  }
+
+  return false;
+}
+
+async function checkAuth(req: NextRequest): Promise<boolean> {
+  // Placeholder authentication logic
+  // You can implement your actual auth check here
+  const authToken = req.headers.get("authorization");
+  console.log({ authToken });
+  return true; // For now, always return true
+}
+
+export function requestHandler<T = Record<string, string | string[]>>(
```

Comment:
need better name & comments, maybe structure as well.

Reply:
Hey @devjiwonchoi , 

I've added some comments. Let me know if that works.